### ISSUE TITLE
Support for Oracle 23ai on Windows Server 2022

### ISSUE THREAD

**Issue by joejan777** [mortenbra/thoth-gateway#10]
Hello - Will Thoth gateway run with Oracle 23ai on a Windows Server 2022 environment?  Looking at some of the old posts it looks like it runs on Oracle 21c on Windows Server 2019.  Thanks - this is a great product.

**Comment by mortenbra** [comment]
Hi, I haven't had the time to test it myself yet. You could try it out and report back if you face any issues? Most upgrades have been fairly painless but sometimes the bundled ODP.NET drivers would need an update as well.

**Comment by joejan777** [comment]
Yes I could do that once I get Oracle 23ai on-prem.  I have been talking to another thoth gateway user on Oracle.com and he has issues with exporting applications and trying to import it.  I had the same issue (Oracle 19c/APEX 24.1) and I fixed it by using Thoth 1.4.4 and granting numerous privs on flow_files.wwv_flow_file_objects$  to apex_public_user and anonymous(maybe overkill on that user).  So thoth gateway is working with the latest APEX 24.1 version (patched).  Thoth gateway is a fantastic application and working  with IIS I cannot get the same functionally using tomcat/ords.  I would love to see it work with Oracle 23ai.  Thanks for this application!

**Comment by mortenbra** [comment]
The situation with the missing privs on wwv_flow_file_object$ is because after Oracle desupported mod_plsql, they changed ORDS so that it no longer inserts uploads into that table directly, but rather calls the htmldb_util.set_clob procedure. The Thoth Gateway should start doing that as well. As time goes by and Oracle develops ORDS further, there may be more such changes under the hood (between APEX and ORDS) that could cause incompatibilities unless the Thoth Gateway is updated accordingly.

**Comment by joejan777** [comment]
Thank you for that explanation.  I will update this once I install Thoth Gateway with Oracle 23ai on Windows Server 2022.  Thanks again for writing Thoth Gateway!
